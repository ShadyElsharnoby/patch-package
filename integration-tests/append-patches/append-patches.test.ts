import { runIntegrationTest } from "../runIntegrationTest"
runIntegrationTest({
  projectName: "append-patches",
  shouldProduceSnapshots: true,
})
