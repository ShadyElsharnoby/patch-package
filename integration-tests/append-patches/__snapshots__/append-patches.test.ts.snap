// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test append-patches: 00: basic patch file 1`] = `
"SNAPSHOT: basic patch file
left-pad+1.3.0.patch
END SNAPSHOT"
`;

exports[`Test append-patches: 01: after appending a patch file 1`] = `
"SNAPSHOT: after appending a patch file
left-pad+1.3.0+001+initial.patch
left-pad+1.3.0+002+MillionDollars.patch
END SNAPSHOT"
`;

exports[`Test append-patches: 02: the second patch file should go from patch-package to a million dollars 1`] = `
"SNAPSHOT: the second patch file should go from patch-package to a million dollars
diff --git a/node_modules/left-pad/index.js b/node_modules/left-pad/index.js
index a409e14..73d2a7c 100644
--- a/node_modules/left-pad/index.js
+++ b/node_modules/left-pad/index.js
@@ -3,7 +3,7 @@
      * and/or modify it under the terms of the Do What The Fuck You Want
      * To Public License, Version 2, as published by Sam Hocevar. See
      * http://www.wtfpl.net/ for more details. */
-'use patch-package';
+'use a million dollars';
 module.exports = leftPad;
 
 var cache = [
END SNAPSHOT"
`;

exports[`Test append-patches: 03: creating a first patch file with --append 1`] = `
"SNAPSHOT: creating a first patch file with --append
left-pad+1.3.0+001+FirstPatch.patch
END SNAPSHOT"
`;

exports[`Test append-patches: 04: the squashed patch file should go from use strict to a million dollars 1`] = `
"SNAPSHOT: the squashed patch file should go from use strict to a million dollars
diff --git a/node_modules/left-pad/index.js b/node_modules/left-pad/index.js
index e90aec3..73d2a7c 100644
--- a/node_modules/left-pad/index.js
+++ b/node_modules/left-pad/index.js
@@ -3,7 +3,7 @@
      * and/or modify it under the terms of the Do What The Fuck You Want
      * To Public License, Version 2, as published by Sam Hocevar. See
      * http://www.wtfpl.net/ for more details. */
-'use strict';
+'use a million dollars';
 module.exports = leftPad;
 
 var cache = [
END SNAPSHOT"
`;

exports[`Test append-patches: 05: after appending a billion dollars 1`] = `
"SNAPSHOT: after appending a billion dollars
left-pad+1.3.0+001+FirstPatch.patch
left-pad+1.3.0+002+BillionDollars.patch
END SNAPSHOT"
`;

exports[`Test append-patches: 06: after updating the appended patch file to a TRILLION dollars 1`] = `
"SNAPSHOT: after updating the appended patch file to a TRILLION dollars
diff --git a/node_modules/left-pad/index.js b/node_modules/left-pad/index.js
index 73d2a7c..f53ea10 100644
--- a/node_modules/left-pad/index.js
+++ b/node_modules/left-pad/index.js
@@ -3,7 +3,7 @@
      * and/or modify it under the terms of the Do What The Fuck You Want
      * To Public License, Version 2, as published by Sam Hocevar. See
      * http://www.wtfpl.net/ for more details. */
-'use a million dollars';
+'use a trillion dollars';
 module.exports = leftPad;
 
 var cache = [
END SNAPSHOT"
`;

exports[`Test append-patches: 07: patch-package fails when a patch in the sequence is invalid 1`] = `
"SNAPSHOT: patch-package fails when a patch in the sequence is invalid
patch-package 0.0.0
• Creating temporary folder
• Installing left-pad@1.3.0 with npm
• Diffing your files with clean files
Failed to apply patch left-pad+1.3.0+001+FirstPatch.patch to left-pad
END SNAPSHOT"
`;

exports[`Test append-patches: 08: --append is not compatible with --create-issue 1`] = `
"SNAPSHOT: --append is not compatible with --create-issue
patch-package 0.0.0
--create-issue is not compatible with --append.
END SNAPSHOT"
`;
